conans/test/functional/generators/premake_test.py

```diff
@@ -0,0 +1,77 @@
+import os
+import textwrap
+import unittest
+
+from nose.plugins.attrib import attr
+
+from conans import load
+from conans.client.tools import chdir, which
+from conans.test.utils.tools import TestClient
+
+
+@attr("slow")
+@attr("premake")
+@unittest.skipIf(which("premake5") is None, "Needs premake5")
+class PremakeGeneratorTest(unittest.TestCase):
+
+    def setUp(self):
+        self.client = TestClient()
+        self.client.run("new test/1.0")
+        self.client.run("create . danimtb/testing")
+        conanfile = textwrap.dedent("""
+        [requires]
+        test/1.0@danimtb/testing
+        
+        [generators]
+        premake
+        """)
+        premake = textwrap.dedent("""
+        include("conanbuildinfo.lua")
+
+        workspace("example")
+            conan_basic_setup()
+
+            project("example")
+            kind "ConsoleApp"
+            language "C++"
+            targetdir = "bin/%{cfg.buildcfg}"
+
+            files{
+                "src/**",
+            }
+
+            filter "configurations:Debug"
+                defines { "DEBUG" }
+                symbols "On"
+
+            filter "configurations:Release"
+                defines { "NDEBUG" }
+                optimize "On"
+        """)
+        hello_cpp = textwrap.dedent("""
```

Comment:
If premake will only generate the `sln` file, we do not need this file.

Reply:
You are right. I was thinking about doing a real build, so maybe I don't need all this. Will take a look 🦃